R-packages/covidcast/R/plot.R

```diff
@@ -566,3 +710,63 @@ plot_line = function(x, range = NULL, title = NULL, params = list()) {
   return(ggplot2::ggplot(aes(x = time_value), data = df) +
          line_layer + ribbon_layer + lim_layer + label_layer + theme_layer)
 }
+
+
+get_pr_centroid = function(map_df){
+  pr_df = map_df %>% dplyr::filter(.$is_pr) %>% sf::st_transform(., 102003)
+  pr_centroid = sf::st_centroid(sf::st_geometry(pr_df))
+  return(pr_centroid)
+}
+
+
+shift_pr = function(map_df, pr_centroid){
+  pr_df = map_df %>% dplyr::filter(.$is_pr)
+  pr_df = sf::st_transform(pr_df, 102003)
+  pr_shift = sf::st_geometry(pr_df) + c(-0.9e+6, 1e+6)
+  pr_df = sf::st_set_geometry(pr_df, pr_shift)
+  r = -16 * pi / 180
+  rotation = matrix(c(cos(r), sin(r), -sin(r), cos(r)), nrow = 2, ncol = 2)
+  pr_rotate = (sf::st_geometry(pr_df) - pr_centroid) * rotation + pr_centroid
+  pr_df = sf::st_set_geometry(pr_df, pr_rotate)
+  pr_df = sf::st_set_crs(pr_df, 102003)
+  return(pr_df)
+}
+
+
+get_alaska_centroid = function(map_df){
+  alaska_df = map_df %>% dplyr::filter(.$is_alaska) %>% sf::st_transform(., 102006)
+  alaska_centroid = sf::st_centroid(sf::st_geometry(alaska_df))
+  return(alaska_centroid)
+}
+
+
+shift_alaska = function(map_df, alaska_centroid){
+  alaska_df = map_df %>% dplyr::filter(.$is_alaska)
+  alaska_df = sf::st_transform(alaska_df, 102006)
+  alaska_scale = (sf::st_geometry(alaska_df) - alaska_centroid) * 0.35 + alaska_centroid
+  alaska_df = sf::st_set_geometry(alaska_df, alaska_scale)
+  alaska_shift = sf::st_geometry(alaska_df) + c(-2e+6, -2.6e+6)
+  alaska_df = sf::st_set_geometry(alaska_df, alaska_shift)
+  alaska_df = sf::st_set_crs(alaska_df, 102003)
+  return(alaska_df)
+}
+
+
+shift_hawaii = function(map_df){
+  hawaii_df = map_df %>% dplyr::filter(.$is_hawaii)
+  hawaii_df = sf::st_transform(hawaii_df, 102007)
+  hawaii_shift = sf::st_geometry(hawaii_df) + c(-1e+6, -2e+6)
+  hawaii_df = sf::st_set_geometry(hawaii_df, hawaii_shift)
+  hawaii_df = sf::st_set_crs(hawaii_df, 102003)
+  return(hawaii_df)
+}
+
+
+shift_main = function(map_df){
+  main_df = map_df %>% dplyr::filter(
+      !.$is_alaska) %>% dplyr::filter(
+        !.$is_hawaii) %>% dplyr::filter(!.$is_pr)
+  main_df = sf::st_transform(main_df, 102003)
+  return(main_df)
+}
+    
```

Comment:
It's asking for a newline here, but perhaps you need to delete the whitespace at the end of this file